proposed/simplecache.md

```diff
@@ -155,32 +167,44 @@ interface CacheInterface
     /**
      * Obtain multiple cache items by their unique keys
      *
-     * @param array|Traversable $keys A list of keys that can obtained in a single operation.
+     * @param array $keys An array of keys that can obtained in a single operation.
      *
-     * @return array An array of key => value pairs. Cache keys that do not exist or are stale will have a value of null.
+     * @return array|Traversable A list of key => value pairs. Cache keys that do not exist or are stale must not be returned.
+     *
+     * @throws InvalidArgumentException
+     *   If any of the keys in $keys are not a legal value a \Psr\Cache\InvalidArgumentException
+     *   MUST be thrown.
      */
-    public function getMultiple($keys);
+    public function getMultiple(array $keys);
 
     /**
      * Persisting a set of key => value pairs in the cache, with an optional TTL.
      *
-     * @param array|Traversable     $items An array of key => value pairs for a multiple-set operation.
+     * @param array                 $items An array of key => value pairs for a multiple-set operation.
      * @param null|int|DateInterval $ttl   Optional. The TTL value of this item. If no value is sent and
      *                                     the driver supports TTL then the library may set a default value
      *                                     for it or let the driver take care of that.
      *
      * @return bool True on success and false on failure
+     *
+     * @throws InvalidArgumentException
+     *   If any of the keys in $keys are not a legal value a \Psr\Cache\InvalidArgumentException
+     *   MUST be thrown.
      */
-    public function setMultiple($items, $ttl = null);
+    public function setMultiple(array $items, $ttl = null);
 
     /**
      * Delete multiple cache items in a single operation
      *
-     * @param array|Traversable $keys The array of string-based keys to be deleted
+     * @param array $keys An array of string-based keys to be deleted
      *
-     * @return void
+     * @return bool True if the item was successfully removed. False if there was an error.
+     *
+     * @throws InvalidArgumentException
+     *   If any of the keys in $keys are not a legal value a \Psr\Cache\InvalidArgumentException
+     *   MUST be thrown.
      */
-    public function deleteMultiple($keys);
+    public function deleteMultiple(array $keys);
```

Comment:
Really? I thought you only wanted it for returns.. I didn't see anyone advocating for Traversable inputs.